src/benchmarks/micro/libraries/System.Text.Json/Serializer/WritePreservedReferences.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+using MicroBenchmarks.Serializers;
+using Newtonsoft.Json;
+
+namespace System.Text.Json.Serialization.Tests
+{
+    [GenericTypeArguments(typeof(LoginViewModel))]
+    [GenericTypeArguments(typeof(Location))]
+    [GenericTypeArguments(typeof(IndexViewModel))]
+    [GenericTypeArguments(typeof(MyEventsListerViewModel))]
+    [GenericTypeArguments(typeof(SimpleListOfInt))]
+    [GenericTypeArguments(typeof(SimpleStructWithProperties))]
+    public class WritePreservedReferences<T>
+    {
+        private T _value;
+        private JsonSerializerOptions _options;
+        private JsonSerializerSettings _settings;
+
+        [GlobalSetup]
+        public void Setup()
+        {
+            _value = DataGenerator.Generate<T>();
+
+            _options = new JsonSerializerOptions { ReferenceHandling = ReferenceHandling.Preserve };
+
+            _settings = new JsonSerializerSettings { PreserveReferencesHandling = PreserveReferencesHandling.All };
+        }
+
+        [BenchmarkCategory(Categories.Libraries, Categories.JSON)]
+        [Benchmark]
+        public string SerializePreserved() => JsonSerializer.Serialize(_value, _options);
+
+        [BenchmarkCategory(Categories.ThirdParty, Categories.JSON)]
+        [Benchmark(Baseline = true)]
```

Comment:
This `Baseline` isn't doing what we would want it to do since this benchmark is generic and instead of comparing 
`SerializePreserved` vs `NewtonsoftSerializePreserved`, we are comparing one model vs another, which isn't helpful.

Even though, I think we should remove the `Newtonsoft` test, it brings up an interesting question on how would we get the baseline comparison we want (even on local runs), if we wanted to? @adamsitnik - any suggestions?

Reply:
Is very likely that what is causing such behavior is that I am running the benchmarks with `--join`, it is not refreshing the baseline when joining, should it?

Here's my command:
`dotnet run -c Release -f netcoreapp5.0 --filter *WritePreservedReferences* --join`